exasol/toolbox/util/git.py

```diff
@@ -0,0 +1,51 @@
+import subprocess  # nosec
+from functools import wraps
+from pathlib import Path
+
+
+def run_command(func):
+    @wraps(func)
+    def wrapper(*args, **kwargs) -> str:
+        command_list = func(*args, **kwargs).split()
+        output = subprocess.run(
+            command_list, capture_output=True, text=True, check=True
+        )  # nosec
+        return output.stdout.strip()
+
+    return wrapper
+
+
+class Git:
+    @staticmethod
+    @run_command
+    def get_latest_tag() -> str:
+        """
+        Get the latest tag from the git repository.
+        """
+        return "git describe --tags --abbrev=0"
+
+    @staticmethod
+    @run_command
+    def read_file_from_tag(tag: str, remote_file: str) -> str:
+        """
+        Read the contents of the specified file `remote_file` at the point in time
+        specified by git tag `tag`.
+        """
+        return f"git cat-file blob {tag}:{remote_file}"
+
+    @staticmethod
+    def copy_remote_file_locally(
```

Comment:
Why would the PTB need this function?

Reply:
This is used by `get_dependencies_from_latest_tag` to download the file to be read in by `get_dependencies` and the classes `PoetryToml.load_from_toml` and `PoetryDependencies`. These currently are set to **read** in files. We could choose to adapt them that the text is parsed first, but as the underlying code for `PoetryDependencies` uses `poetry show` which we cannot control, it seemed like as it was easier to copy the file locally in a temporary folder.